pkg/proxy/userspace/roundrobin.go

```diff
@@ -81,22 +81,22 @@ func NewLoadBalancerRR() *LoadBalancerRR {
 	}
 }
 
-func (lb *LoadBalancerRR) NewService(svcPort proxy.ServicePortName, affinityType api.ServiceAffinity, ttlMinutes int) error {
+func (lb *LoadBalancerRR) NewService(svcPort proxy.ServicePortName, affinityType api.ServiceAffinity, ttlSeconds int) error {
 	glog.V(4).Infof("LoadBalancerRR NewService %q", svcPort)
 	lb.lock.Lock()
 	defer lb.lock.Unlock()
-	lb.newServiceInternal(svcPort, affinityType, ttlMinutes)
+	lb.newServiceInternal(svcPort, affinityType, ttlSeconds)
 	return nil
 }
 
 // This assumes that lb.lock is already held.
-func (lb *LoadBalancerRR) newServiceInternal(svcPort proxy.ServicePortName, affinityType api.ServiceAffinity, ttlMinutes int) *balancerState {
-	if ttlMinutes == 0 {
-		ttlMinutes = 180 //default to 3 hours if not specified.  Should 0 be unlimited instead????
+func (lb *LoadBalancerRR) newServiceInternal(svcPort proxy.ServicePortName, affinityType api.ServiceAffinity, ttlSeconds int) *balancerState {
+	if ttlSeconds == 0 {
+		ttlSeconds = int(api.ServiceAffinityClientIPDefaultSeconds) //default to 3 hours if not specified.  Should 0 be unlimited instead????
```

Comment:
> do we need this?

```
if ttlSeconds == 0 {
	ttlSeconds = int(api.ServiceAffinityClientIPDefaultSeconds) //default to 3 hours if not specified.  Should 0 be unlimited instead????
```
I'm afraid we may need this.

Since the function `func (lb *LoadBalancerRR) newServiceInternal()` will be called by OnEndpointsAdd(), see https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/userspace/roundrobin.go#L278-L282

and `OnEndpointsUpdate()`, see https://github.com/kubernetes/kubernetes/blob/master/pkg/proxy/userspace/roundrobin.go#L312-L316

They both set "0" as the ttlSeconds for `newServiceInternal()`